core/src/smartcontracts/isi/triggers.rs

```diff
@@ -35,6 +35,16 @@ pub mod isi {
                 }
             }
 
+            if let Some(ref domain_id) = new_trigger.action.domain {
+                let domain = wsv.domain(domain_id)?;
+
+                let trigger_id = new_trigger.id.clone();
+
+                wsv.world
+                    .domain_scoped_triggers
+                    .insert(trigger_id, domain.id().clone());
```

Comment:
Let's not use `world` from `WorldStateVeiw` directly. Be design all instructions should try to use only `modify_*` methods to modify something inside wsv.

Reply:
Also I see, that your domain-scoped triggers are stored in World's `TriggerSet` as ordinary triggers, because of the old code below.
That won't help us to increase performance for events filtering

Reply:
So filter of domain-event should not try to match events from different domains (like global triggers). But right now it's working like global trigger.
And, I think, it should be fixed not in filters but in the data flow

Reply:
@Arjentix I've added the "matching" trigger's domain to the event's domain before adding actions to "matched_ids". It will allow avoiding "blocking_write" and extrа iterations over "matched_ids". What do you think about this approach?